backend/main.py

```
@@ -1,306 +1,136 @@
-
-if __name__ == "__main__":
-    import uvicorn
-    uvicorn.run("main:app", host="0.0.0.0", port=8000, reload=True)
\ No newline at end of file
+@app.get("/api/search")
+async def search_drawings(q: str, limit: int = 10):
+    async with httpx.AsyncClient(timeout=15.0) as client:
+        try:
+            response = await client.post(f"{AGENT_URL}/search", json={"query": q, "limit": limit})
+            return response.json()
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)